### DISCUSSION TITLE
Can DeePMD potentials be used in LAMMPS along with classical empirical potentials?

### DISCUSSION THREAD

**Discussion by kuntalg97** [deepmodeling/deepmd-kit#2984] [Q&A]
I used DeePMD for generating a NN potential in the .pb format. And I can successfully run MD simulations using this potential, only when the whole system is described by the DeePMD generated potential. I'm using LAMMPS.

However, I'm looking at a specific case where I need part of my system to be described by the DeePMD potential, and the remainder to be described by standard CHARMM/OPLS force fields (similar to QM/MM except instead of QM it'll be the DeePMD potential). I would also like to use the DeePMD potential in conjunction with tabulated pair potentials.

I was trying to use commands of the following type :

pair_style          hybrid deepmd graph.pb lj/cut/coul/long 10.0 10.0
pair_style          hybrid/overlay deepmd graph.pb lj/cut/coul/long 10.0 10.0
pair_style          hybrid deepmd graph.pb table spline 2000
pair_style          hybrid/overlay deepmd graph.pb table spline 2000

However, I'm encountering numerous errors including illegal pair style / issues with identifying pair coefficients with the DeePMD potential, etc. This includes the following error:

ERROR: deepmd requires that the scale should be set to all atom types, i.e. pair_coeff * *.

Does the current implementation of DeePMD in LAMMPS support such a hybrid scheme? Or can it be used ONLY when the entire system is described by the DeePMD potential?

**Comment by njzjz** [comment]
According to LAMMPS documentation, `hybrid/overlay` should be used instead of `hybrid`.

**Reply by kuntalg97** [reply]
I did. Now the error says :

ERROR: deepmd requires that the scale should be set to all atom types, i.e. pair_coeff * *.

**Reply by njzjz** [reply]
As is shown in the error message, `pair_coeff * *` should be used. Set `NULL` for elements that are not described, e.g. `pair_coeff * * NULL O H NULL`.

**Reply by kuntalg97** [reply]
Hi, say my DeePMD model has two types denoted by O and O1 (corresponding to atom types 1 and 2 in LAMMPS). I wish to use DeePMD for type O only, and tabulated pair potentials for the O-O1 and O1-O1 interactions (and ignore the DeePMD part of O1). I'm using the following snippet:

pair_style      hybrid/overlay deepmd graph_two.pb table spline 2000
pair_coeff      * *  deepmd  O NULL
pair_coeff      1 2  table   pair_1_2.table  1_2
pair_coeff      2 2  table   pair_2_2.table  2_2

After a few runs, I get the following error:

/DeePMD/deepmd/bin/lmp: line 11: 2554595 Floating point exception(core dumped) .../DeePMD/deepmd/bin/_lmp "$@"

Where am I going wrong?

**Reply by njzjz** [reply]
It might be a previous bug https://github.com/deepmodeling/deepmd-kit/issues/2668
